pydy/system.py

```diff
@@ -340,22 +341,26 @@ def times(self):
         equations of motion are integrated, numerically.
 
         The object should be in a format which the integration module to be
-        used can accept. Since this attribute is not checked for
-        compatibility, the user becomes responsible to supply it correctly.
+        used can accept.
         """
         return self._times
 
     @times.setter
     def times(self, new_times):
+        self._check_times(new_times)
         self._times = new_times
 
     def _check_times(self, times):
-        """
-        Very basic checking.
-        TODO: add more checking
-        """
         if len(times) == 0:
             raise TypeError("Times supplied should be in an array_like format.")
+
+        times = np.array(times)
```

Comment:
I think asarray is the typical choice for this, see:

http://stackoverflow.com/questions/14415741/numpy-array-vs-asarray